test/Microsoft.IdentityModel.JsonWebTokens.Tests/JwtTokenUtilitiesTests.cs

```diff
@@ -208,6 +215,281 @@ public void ResolveTokenSigningKey()
+        {
+            // Arrange
+            // SecurityKey can never have 'null' CryptoProviderFactory.
+            var securityKey = symmetricSecurityKey;
+            var validationParameters = new TokenValidationParameters()
+            {
+                CryptoProviderFactory = new CryptoProviderFactory(),
+            };
+
+            // Act
+            var result = JwtTokenUtilities.s_getCryptoProviderFactory(validationParameters, securityKey);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Same(validationParameters.CryptoProviderFactory, result);
+        }
+
+        [Fact]
+        public void GetCryptoProviderFactory_WhenSetToNullOnTokenValidationParameters_ReturnsFromSecurityKey()
+        {
+            // Arrange
+            // SecurityKey can never have 'null' CryptoProviderFactory.
+            var securityKey = symmetricSecurityKey;
+            var validationParameters = new TokenValidationParameters
+            {
+                CryptoProviderFactory = null,
+            };
+
+            // Act
+            var result = JwtTokenUtilities.s_getCryptoProviderFactory(validationParameters, securityKey);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(securityKey.CryptoProviderFactory, result);
+        }
+
+        [Fact]
+        public void DecryptJwtToken_WhenValidationParametersIsNull_ShouldThrow()
+        {
+            // Arrange
+            TokenValidationParameters validationParameters = null;
+
+            // Act & Assert
+            var exception = Assert.Throws<ArgumentNullException>(() =>
+                JwtTokenUtilities.DecryptJwtToken(
+                    securityToken: null,
+                    validationParameters: validationParameters,
+                    decryptionParameters: null));
+
+            Assert.Equal("validationParameters", exception.ParamName);
+        }
+
+        [Fact]
+        public void DecryptJwtToken_WhenJwtTokenDecryptionParametersIsNull_ShouldThrow()
+        {
+            // Arrange
+            var validationParameters = new TokenValidationParameters();
+            JwtTokenDecryptionParameters decryptionParameters = null;
+
+            // Act & Assert
+            var exception = Assert.Throws<ArgumentNullException>(() =>
+                JwtTokenUtilities.DecryptJwtToken(
+                    securityToken: null,
+                    validationParameters: validationParameters,
+                    decryptionParameters: decryptionParameters));
+
+            Assert.Equal("decryptionParameters", exception.ParamName);
+        }
+
+        [Fact(Skip = "Skip due to potential false positives it may cause running tests in parallel as it overrides JwtTokenUtilities.s_getCryptoProviderFactory.")]
+        public void DecryptJwtToken_WhenCryptoProviderFactoryIsNull_FailsWithIDX10609AndLogsIDX10607()
```

Comment:
nit: I think for the naming just a general `_ThrowsException` is okay. It's a bit easier to read and we can look down in asserts what specific exception is thrown.
```suggestion
        public void DecryptJwtToken_WhenCryptoProviderFactoryIsNull_ThrowsException()
```

Reply:
Usually, I use the general ``_ThrowsExpcetion`` too; somehow I didn't here. Updated.